api/src/main/java/org/infinispan/api/mutiny/MutinyCache.java

```diff
@@ -51,132 +53,366 @@
+      return keys(CacheOperationOptions.DEFAULT);
+   }
+
+   /**
+    * Retrieve all keys
+    *
+    * @return @{@link Multi} which produces keys as items.
+    */
+   Multi<K> keys(CacheOperationOptions options);
+
+   /**
+    * Retrieve all entries
+    *
+    * @return
+    */
+   default Multi<CacheEntry<K, V>> entries() {
+      return entries(CacheOperationOptions.DEFAULT);
+   }
 
    /**
     * Retrieve all entries
     *
+    * @param options
+    * @return
+    */
+   Multi<CacheEntry<K, V>> entries(CacheOperationOptions options);
+
+   /**
+    * Retrieve all the entries for the specified keys.
+    *
+    * @param keys
     * @return
     */
-   Multi<? extends CacheEntry<K, V>> entries();
+   default Multi<CacheEntry<K, V>> getAll(Set<K> keys) {
+      return getAll(keys, CacheOperationOptions.DEFAULT);
+   }
 
    /**
     * Retrieve all the entries for the specified keys.
     *
     * @param keys
+    * @param options
     * @return
     */
-   Multi<? extends CacheEntry<K, V>> getMany(List<K> keys);
+   Multi<CacheEntry<K, V>> getAll(Set<K> keys, CacheOperationOptions options);
 
-   Multi<? extends CacheEntry<K, V>> getMany(K... keys);
+   /**
+    * @param keys
+    * @return
+    */
+   default Multi<CacheEntry<K, V>> getAll(K... keys) {
+      return getAll(CacheOperationOptions.DEFAULT, keys);
+   }
+
+   /**
+    * @param keys
+    * @param options
+    * @return
+    */
+   Multi<CacheEntry<K, V>> getAll(CacheOperationOptions options, K... keys);
 
    /**
     * Put multiple entries from a {@link Multi}
     *
     * @param pairs
     * @return Void
     */
-   Multi<MutinyWriteResult<K>> put(Multi<CacheEntry<K, V>> pairs);
+   default Multi<MutinyWriteResult<K>> put(Multi<CacheEntry<K, V>> pairs) {
+      return put(pairs, CacheWriteOperationOptions.DEFAULT);
+   }
+
+   /**
+    * @param pairs
+    * @param options
+    * @return
+    */
+   Multi<MutinyWriteResult<K>> put(Multi<CacheEntry<K, V>> pairs, CacheWriteOperationOptions options);
+
+   /**
+    * @param pairs
+    * @return
+    */
+   default Multi<MutinyWriteResult<K>> put(List<CacheEntry<K, V>> pairs) {
+      return put(pairs, CacheWriteOperationOptions.DEFAULT);
+   }
 
-   Multi<MutinyWriteResult<K>> put(List<CacheEntry<K, V>> pairs);
+   /**
+    * @param pairs
+    * @param options
+    * @return
+    */
+   Multi<MutinyWriteResult<K>> put(List<CacheEntry<K, V>> pairs, CacheWriteOperationOptions options);
 
-   Multi<MutinyWriteResult<K>> put(Map<K, V> map, CacheEntryMetadata metadata);
+   /**
+    * @param map
+    * @param options
+    * @return
+    */
+   Multi<MutinyWriteResult<K>> put(Map<K, V> map, CacheWriteOperationOptions options);
+
+   default Multi<MutinyWriteResult<K>> put(Map<K, V> map) {
+      return put(map, CacheWriteOperationOptions.DEFAULT);
+   }
+
+   /**
+    * Removes a set of keys. Returns the keys that were removed.
+    *
+    * @param keys
+    * @return
+    */
+   default Multi<K> removeAll(Multi<K> keys) {
```

Comment:
missing `removeAll(Set)`. probably missing `AsyncCache.removeAll(Flowable)` (or `removeAll(Stream)`) to keep the interfaces in sync.

Reply:
no  `replace` either.

Reply:
Done